python/shotgun_model/data_handler_nav.py

```diff
@@ -146,10 +152,40 @@ class and the data payload from Shotgun should be provided via the
         num_modifications = 0
 
         new_uids = set()
+
+        # a list of sg data dicts to display items for
+        child_data = []
+
+        if item_path == self.__root_path:
+            self._log_debug("This is the root of the tree.")
+            parent_uid = None
+
+            if self.__include_root:
```

Comment:
Totally on board with your concerns. Search box, asset expansion, home button all appear to be working fine. The home button takes you to the current context. If the context is the project, it takes you to the unselected state which is what I expect. 

Regarding the path, I was originally removing it all together from the special item and it was working fine. Felt a bit unnecessary, but maybe it should be that way.